src/kagglehub/clients.py

```diff
@@ -56,6 +94,8 @@ def __init__(self) -> None:
         self.credentials = get_kaggle_credentials()
         self.endpoint = get_kaggle_api_endpoint()
 
+        self.user_agent_manager = KaggleHubUserAgent()
```

Comment:
A new KaggleApiV1Client is created before each call, this means it will create a new KaggleHubUserAgent every time and won't benefit from any caching. Hence the suggestion to use a global variable instead with a simple function.